os/net/lwip/src/api/sockets.c

```diff
@@ -2055,6 +2075,7 @@ int lwip_getsockopt(int s, int level, int optname, void *optval, socklen_t *optl
 	LWIP_SETGETSOCKOPT_DATA_VAR_REF(data).level = level;
 	LWIP_SETGETSOCKOPT_DATA_VAR_REF(data).optname = optname;
 	LWIP_SETGETSOCKOPT_DATA_VAR_REF(data).optlen = *optlen;
+	LWIP_SETGETSOCKOPT_DATA_VAR_REF(data).pid = getpid();
```

Comment:
you have to consider following case.
there are Task1(T1) Task2(T2). and they have same socket S1
T1 calls getsockopt, T2 calls setsockopt then calling LWIP_SETGETSOCKOPT_DATA_VAR_REF() can be mixed because of context switching. I think you can solve this problem regarding err_t tcpip_send_msg_wait_sem(tcpip_callback_fn fn, void *apimsg, sys_sem_t *sem); Check op_sync

Reply:
can you please clear it more "I think you can solve this problem regarding err_t tcpip_send_msg_wait_sem(tcpip_callback_fn fn, void *apimsg, sys_sem_t *sem); Check op_sync" ??

Reply:
for example T1, T2 use same netconn because they access same socket, lwIP sends signal to both tasks with same netconn->op_complete.  To prevent this op_sync is added.
I think same things could be heppend to lwip_setsockopt too.
Or you can consider LWIP_NETCONN_SEM_PER_THREAD option which was not defined when op_sync is added.
I found it recently.